### PR TITLE
PRESS4-267 Analytics for ECommerce Module

### PR DESCRIPTION
**Pull request by sumathi3399**
JIRA- https://jira.newfold.com/browse/PRESS4-267

- Integrated hive analytics in Ecommerce module.
- Added [js-utility-ui-analytics](https://github.com/newfold-labs/js-utility-ui-analytics) package for sending the event logs.
- For sending the single event we will be hitting **`/newfold-data/v1/events`** api.
- There will be four parameters while sending the HiiveEvent (category, action, data, namespace).

Screenshot 2023-07-03 at 3 40 45 PM

Screenshot 2023-07-03 at 3 40 11 PM

Screenshot 2023-07-03 at 3 40 23 PM

### CODE REVIEW

File: src/utils/analytics/index.js
Comment (aulisius) [inline src/utils/analytics/index.js:1]: Can we move into `sdk` folder and use it as `AnalyticsSdk.track`

File: src/index.js [hunk 1]
```diff
@@ -7,6 +7,23 @@ import { StoreDetails } from "./components/StoreDetails";
 import { WonderCart } from "./components/WonderCart";
 import { createApiUrl } from "./sdk/createApiUrl";
 import { PluginsSdk } from "./sdk/plugins";
+import { HiiveAnalytics } from '@newfold-labs/js-utility-ui-analytics';
+import domReady from '@wordpress/dom-ready';
```
Comment (aulisius) [inline src/index.js:11]: Can we add `@wordpress/dom-ready` to dev dependencies?

File: src/index.js [hunk 2]
```diff
@@ -7,6 +7,23 @@ import { StoreDetails } from "./components/StoreDetails";
...
+
+domReady(() => {
+  HiiveAnalytics.initialize({
```
Comment (aulisius) [inline src/index.js:14]: Can we move the initialize function into the `sdk` as commented below

**Review by aulisius — CHANGES_REQUESTED** [review]

File: src/utils/analytics/index.js
Comment (sumathi3399) [reply src/utils/analytics/index.js:1]: done

File: src/index.js (hunk 1 above)
Comment (sumathi3399) [reply src/index.js]: done

File: src/index.js (hunk 2 above)
Comment (sumathi3399) [reply src/index.js:14]: kept as it is. as this method needs to initialise when the application starts.

File: src/sdk/analytics.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+} from "@newfold-labs/js-utility-ui-analytics";
+
+export const AnalyticsSdk = {
+  track(action, value) {
+    const hiiveEvent = new HiiveEvent(
+      "wp-module-ecommerce",
+      action,
+      {
+        value,
+        timestamp: Date.now(),
```
Comment (aulisius) [inline src/sdk/analytics.js:13]: Add location information

Comment (sumathi3399) [reply src/sdk/analytics.js:14]: Added the location property

**Pull request closed without merging**